File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/base/constraints/regular/RegularDialogConstraints.java
package com.github.RooneyAndShadows.lightbulb.dialogs.base.constraints.regular;

public class RegularDialogConstraints {
    private final int minWidth;
    private final int maxWidth;
    private final int minHeight;
    private final int maxHeight;

    public RegularDialogConstraints(int minWidth, int maxWidth, int minHeight, int maxHeight) {
        this.minWidth = minWidth;
        this.maxWidth = maxWidth;
        this.minHeight = minHeight;
        this.maxHeight = maxHeight;
    }

    public int resolveWidth(int desiredWidth) {
        if (desiredWidth > getMaxWidth())
            desiredWidth = getMaxWidth();
        if (desiredWidth < getMinWidth())
            desiredWidth = getMinWidth();
        return (desiredWidth);
    }

    public int resolveHeight(int desiredHeight) {
        if (desiredHeight > getMaxHeight())
            desiredHeight = getMaxHeight();
        if (desiredHeight < getMinHeight())
            desiredHeight = getMinHeight();
        return (desiredHeight);
    }

    public int getMinWidth() {
        return Math.min(minWidth, maxWidth);
    }

    public int getMaxWidth() {
        return Math.max(maxWidth, minWidth);
    }

    public int getMinHeight() {
        return Math.min(minHeight, maxHeight);
    }

    public int getMaxHeight() {
        return Math.max(maxHeight, minHeight);
    }
}

File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/base/constraints/regular/RegularDialogConstraints.kt
package com.github.rooneyandshadows.lightbulb.dialogs.base.constraints.regular

import kotlin.math.max
import kotlin.math.min

class RegularDialogConstraints(
    private val minWidth: Int,
    private val maxWidth: Int,
    private val minHeight: Int,
    private val maxHeight: Int
) {
    fun resolveWidth(desiredWidth: Int): Int {
        var result = desiredWidth
        if (result > getMaxWidth()) result = getMaxWidth()
        if (result < getMinWidth()) result = getMinWidth()
        return result
    }

    fun resolveHeight(desiredHeight: Int): Int {
        var result = desiredHeight
        if (result > getMaxHeight()) result = getMaxHeight()
        if (result < getMinHeight()) result = getMinHeight()
        return result
    }

    fun getMinWidth(): Int {
        return min(minWidth, maxWidth)
    }

    fun getMaxWidth(): Int {
        return max(maxWidth, minWidth)
    }

    fun getMinHeight(): Int {
        return min(minHeight, maxHeight)
    }

    fun getMaxHeight(): Int {
        return max(maxHeight, minHeight)
    }
}